packages/app/src/components/DBSessionPanel.tsx

```diff
@@ -0,0 +1,139 @@
+import { useMemo } from 'react';
+import { TSource } from '@hyperdx/common-utils/dist/types';
+import { Loader } from '@mantine/core';
+
+import SessionSubpanel from '@/SessionSubpanel';
+import { useSource } from '@/source';
+import { getDisplayedTimestampValueExpression } from '@/source';
+
+import { useEventsData } from './DBTraceWaterfallChart';
+
+export const useSessionId = ({
+  sourceId,
+  traceId,
+  dateRange,
+  enabled = false,
+}: {
+  sourceId?: string;
+  traceId: string;
+  dateRange: [Date, Date];
+  enabled?: boolean;
+}) => {
+  const { data: source } = useSource({ id: sourceId });
+
+  const config = useMemo(() => {
+    if (!source) {
+      return;
+    }
+    return {
+      select: [
+        {
+          valueExpression: getDisplayedTimestampValueExpression(source),
```

Comment:
style: better to centralize getDisplayedTimestampValueExpression call somewhere top level. for now, I think we can use `timestampValueExpression` directly